TS_1/task2.1.ts

```diff
@@ -0,0 +1,74 @@
+// 1. Создайте интерфейс IEmployee с полями name, surname, salary, address (типы для этих полей такие же как в ItEmployee из таск 1)
+//   Создайте функцию getEmployeeInfo(employee), выводящую в консоль всю информацию про employee (формат текста придумать самим)
+//   Создайте type guard isItEmployee, принимающий юнион типов IEmployee и ItEmployee. Используйте его в функции getEmployeeInfo. 
+//   Если на входе itEmployee - выводите в консоль все поля айтишника (а не только те, что в employee)
+//   Функция должна принимать union type между IEmployee и ItEmployee, и через тайпгвард определять что за объект и как работать с ним 
+
+
+interface ItEmployee {
+    name: string;
+    surname: string;
+    readonly salary: number;
+    grade: Grade;
+    address?: IAddress;
+    projectNames: string[];
+}
+
+type Grade = "junior" | "middle" | "senior" | "lead";
+interface IEmployee {
+    name: string;
+    surname: string;
+    readonly salary: number;
+    address?: IAddress;
+    projectNames: string[];
+}
+ 
+interface IAddress {
+    country: string;
+    street: string;
+    house: number;
+    flat: number;
+}
+
+function isItEmployee(obj: object): obj is ItEmployee {
+    return "grade" in obj;
+}
+
+function getEmployeeInfo(employeeOrNot: IEmployee | ItEmployee): void {
+    if (isItEmployee(employeeOrNot)) {
+        const itEmployee = employeeOrNot as ItEmployee;
```

Comment:
Why do you need this variable with type casting ? 
`employeeOrNot` alredy has `ItEmployee` type after using type guard
same for the 48 line below